### PR TITLE
Iteration methods for CompoundTag

### PR DESCRIPTION
**Pull request by SpecificProtagonist**
Adds methods for iterating over a `CompoundTag` (as well as constructing it from a iterator over `(&str, Tag)`). 
Previously it wasn't possible to list what tags a `CompoundTag` contains.   
Alternatively to this PR `CompoundTag.tags` could be made `pub`, but that would make the dependency on linked-hash-map
part of the public API and also demand slightly more boilerplate from the user.

### CODE REVIEW

**Comment by vaIgarashi** [comment]
Hi! Thanks for your pull request. Can you provide example where this may be needed?

**Comment by SpecificProtagonist** [comment]
When building something akin to NBTExplorer, where the structure of the nbt isn't known beforehand (of course, in the majority of other use-cases, the shape of the data is known and this feature doesn't matter).

File: src/lib.rs [hunk 1]
```diff
@@ -633,3 +684,45 @@ fn test_contains_key() {
...
+        }
+    }
+
+    // Test into_iter
+    {
+        let mut iter = compound.into_iter().map(|(name, tag)|
+            (name, match tag { Tag::Int(value) => value, _ => panic!() })
+        );
+        assert_eq!(iter.next(), Some((String::from("test1"), 1)));
+        assert_eq!(iter.next(), Some((String::from("test2"), 10)));
```
Comment (vagola) [inline src/lib.rs:724]: This one was tricky :)

**Review by vaIgarashi — APPROVED** [review]

**Comment by vaIgarashi** [comment]
Thanks for contribution!

**Comment by SpecificProtagonist** [comment]
No problem.

**Pull request merged**